lib/src/css_mirroring/bidi_css_generator.dart

```diff
@@ -0,0 +1,134 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+library scissors.src.css_mirroring.bidi_css_generator;
+
+import 'dart:async';
+
+import 'package:csslib/parser.dart' show parse;
+import 'package:csslib/visitor.dart'
+    show
+        RuleSet,
+        StyleSheet,
+        TreeNode,
+        Declaration,
+        Directive,
+        MediaDirective,
+        HostDirective,
+        PageDirective,
+        CharsetDirective,
+        FontFaceDirective,
+        ImportDirective,
+        NamespaceDirective;
+import 'package:quiver/check.dart';
+import 'package:source_maps/refactor.dart';
+import 'package:source_span/source_span.dart';
+
+import '../utils/enum_parser.dart';
+import 'transformer.dart' show CssMirroringSettings, Direction, flipDirection;
+
+part 'directives_processor.dart';
+part 'rulesets_processor.dart';
+part 'util_classes.dart';
+part 'util_functions.dart';
+
+/// Input CSS flipper function from css mirroring.
+typedef Future<String> CssFlipper(String inputCss);
+
+/// BidiCssGenerator generates a CSS which comprises of orientation neutral,
+/// orientation specific and flipped orientation specific parts.
+///
+/// See README.md for more details.
+
+class BidiCssGenerator {
+  final String _originalCss;
+  final String _flippedCss;
+  final String _cssSourceId;
+  final Direction _nativeDirection;
+
+  TopLevelsPair _topLevels;
+  DirectionIndependentDirectivesProcessor dirIndependentProcessor;
+  DirectionDependentDirectivesProcessor dirDependentProcessor ;
+  RuleSetsProcessor ruleSetsProcessor ;
+
+  BidiCssGenerator._(this._originalCss, this._flippedCss, this._cssSourceId,
+      this._nativeDirection) {
+
+    _topLevels = new TopLevelsPair(
+        parse(_originalCss).topLevels,
+        parse(_flippedCss).topLevels
+    );
+
+     dirIndependentProcessor = new DirectionIndependentDirectivesProcessor();
```

Comment:
these instances are useless, please drop the fields + the class and just use functions